src/arrays/constructors.jl

```diff
@@ -16,4 +16,17 @@
     one_at_ind!(arr, i) = setindex!(arr, one(eltype(arr)), i)
     single_coeff(i, lens...) = one_at_ind!(zeros(lens), i)
 
-export statevec
\ No newline at end of file
+function coherentstatevec(n::Int,alpha::Number;method="direct")
+    s = zeros(n)
```

Comment:
This is actually more tricky. If you pass a complex `alpha` you will get `InexactError` since `s` is an array of floats. I think using `zeros(typeof(float(alpha)),n)` might work (`float` is necessary because you might input an `Int`, but the coefficients are clearly not integer).

However, I am not so happy with having two functions `coherentstatevec` differing only in the presence of a keyword. Either you merge them into one function and pick the method according to the keyword or you rename this one to `coherentstatevec_inf` or such and don't export it. I vote for the latter, since this function strictly only works for infinite `n` and one might get hard to track errors if one uses it for smallish `n`.

Moreover, please add more comments. I know that we are/have been very sloppy about this, but in particular for the "physics-related" stuff (special operators and states etc) I think we should add information about which formulas we are implementing, possibly with a reference. Already in the near future we will start to scratch our heads and wonder what we actually did a couple of months back.